controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -130,3 +142,47 @@ func configureServiceMeshFeatures() feature.FeaturesProvider {
 		return nil
 	}
 }
+
+func IsDefaultAudiences(specAudiences *[]string) bool {
+	return specAudiences == nil || reflect.DeepEqual(*specAudiences, defaultAudiences)
+}
+
+// DefinedAudiencesOrDefault returns the default audiences if the provided audiences are default, otherwise it returns the provided audiences.
+func DefinedAudiencesOrDefault(specAudiences *[]string) []string {
+	if IsDefaultAudiences(specAudiences) {
+		return fetchClusterAudiences()
+	}
+	return *specAudiences
+}
+
+func fetchClusterAudiences() []string {
+	restCfg, err := config.GetConfig()
+	if err != nil {
+		smSetupLog.Error(err, "Error getting config, using default audiences")
+		return defaultAudiences
+	}
+
+	tokenReview := &authentication.TokenReview{
+		Spec: authentication.TokenReviewSpec{
+			Token: restCfg.BearerToken,
```

Comment:
Instead of wiring restCfg again, assuming we are inside of the pod, we could probably just read SA token from `/var/run/secrets/kubernetes.io/serviceaccount/token`. Would that work?

Reply:
Done, added a GetSAToken() func to the cluster package

e38a8ad